elements/lisk-p2p/src/peer/outbound.ts

```diff
@@ -165,6 +165,10 @@ export class OutboundPeer extends Peer {
 			});
 		});
 
+		outboundSocket.on('ping', () => {
+			this.emit('pong', Date.now());
```

Comment:
We cannot trust the time which the peer provides in the pong event which they emit. We need to calculate the delay ourselves; start the timer when we do the ping and stop the timer when we receive the pong from the peer.